aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -69,33 +71,48 @@
 
     private AWSDataStorePlugin(
             @NonNull ModelSchemaRegistry modelSchemaRegistry,
-            @NonNull ModelProvider modelProvider) {
+            @NonNull ModelProvider modelProvider,
+            @NonNull GraphQlBehavior api) {
         this.sqliteStorageAdapter = SQLiteStorageAdapter.forModels(modelSchemaRegistry, modelProvider);
         this.storageItemChangeConverter = new GsonStorageItemChangeConverter();
-        this.orchestrator = createOrchestrator(modelProvider, modelSchemaRegistry, sqliteStorageAdapter);
         this.categoryInitializationsPending = new CountDownLatch(1);
-    }
-
-    private Orchestrator createOrchestrator(
-            ModelProvider modelProvider, ModelSchemaRegistry modelSchemaRegistry, LocalStorageAdapter storageAdapter) {
-        return new Orchestrator(
+        this.orchestrator = new Orchestrator(
             modelProvider,
```

Comment:
Yay for getting rid of useless helper methods and consolidating code in one place!

Reply:
When I first created the factory method, it had some actual logic, to create several instances from only a few dependencies.

But as the code got refactored a few times, the argument list grew long, and yea -- the factory was only hurting the readability.